### PR TITLE
Remove Python2 crumbs

### PR DESCRIPTION
**Pull request by a-detiste**
This is just a little cleanup.

The outdated docs might be confusing to newcomers who didn't got to live through the great Py2 -> 3 transition.

Greetings

### CODE REVIEW

File: src/twisted/words/xish/xpathparser.py [hunk 1]
```diff
@@ -43,7 +43,6 @@
 
 exec(
     r'''
-from __future__ import print_function
```
Comment (adiroiban) [inline src/twisted/words/xish/xpathparser.py:46]: There is this comment at the top of the file 

```
# DO NOT EDIT xpathparser.py!
```

... I guess that this is ok... we don't need to re-run Yapps for this cleanup

**Review by adiroiban — APPROVED** [review]
Thanks for the clenaup. It looks good.

I have added a newsfragment file to help with generating the future release notes.

I think that this is a small cleanup and we don't need to add more info for the release notes

File: src/twisted/words/xish/xpathparser.py (hunk 1 above)
Comment (a-detiste) [reply src/twisted/words/xish/xpathparser.py:46]: I think it's fine if the two copies are updated in sync.

Comment (a-detiste) [reply src/twisted/words/xish/xpathparser.py:46]: No need for release notes for this.

**Comment by adiroiban** [comment]
I have merged this as this is mostly a cleanup.

Thanks Alexandre for your help.

---------

@itamarst the performance tests are quite flaky 

For now I have just merged this PR by ignoring their results.

**Pull request merged**